src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -0,0 +1,154 @@
+     *
+     * Subclasses can override to provide their own filters
+     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter} composition methods.
+     */
+    public CountingReadFilter makeReadFilter(){
+        return new CountingReadFilter("Wellformed", new WellformedReadFilter(getHeaderForReads()))
+                .and(new CountingReadFilter("Mapped", ReadFilterLibrary.MAPPED));
+    }
+
+    /**
+     * Get the information about how to downsample the reads. By default {@link org.broadinstitute.hellbender.utils.downsampling.DownsamplingMethod#NONE} is returned.
+     * Subclasses should override it to provide own downsampling methods.
+     *
+     * @return the downsampling method for the reads
+     */
+    public DownsamplingMethod getDownsamplingMethod() {
+        // TODO: change when downsampling command line options are implemented
+        return DownsamplingMethod.NONE;
+    }
+
+    /**
+     * Marked final so that tool authors don't override it. Tool authors should override onTraversalStart() instead.
+     */
+    @Override
+    protected final void onStartup() {
+        super.onStartup();
+        if ( hasIntervals() ) {
+            reads.setIntervalsForTraversal(intervalsForTraversal);
+        }
+    }
+
+    /**
+     * Implementation of locus-based traversal.
+     * Subclasses can override to provide their own behavior but default implementation should be suitable for most uses.
+     *
+     * The default implementation iterates over all positions in the reference for all samples in the read groups, using
+     * the downsampling method provided by {@link #getDownsamplingMethod()}
+     * and including deletions if {@link #includeDeletions()} returns {@code true}.
+     *
+     * {@link #intervalsForTraversal} are not used in this implementation
+     */
+    @Override
+    public void traverse() {
+        final SAMFileHeader header = getHeaderForReads();
+        // get the samples from the read groups
+        final Set<String> samples = header.getReadGroups().stream()
+                                          .map(SAMReadGroupRecord::getSample)
+                                          .collect(Collectors.toSet());
+        CountingReadFilter countedFilter = disableAllReadFilters ?
+                new CountingReadFilter("Allow all", ReadFilterLibrary.ALLOW_ALL_READS ) :
+                makeReadFilter();
+        // TODO: wrap reads.iterator() with ReadFilteringIterator and countedFilter
+        // get the LIBS
+        LocusIteratorByState libs = new LocusIteratorByState(new ReadFilteringIterator(reads.iterator(), countedFilter), getDownsamplingMethod(), includeDeletions(), KEEP_UNIQUE_READ_LIST_IN_LIBS, samples, header);
+		// prepare the iterator
+		Spliterator<AlignmentContext> iterator = (hasIntervals()) ? new IntervalOverlappingIterator(libs, intervalsForTraversal).spliterator() : libs.spliterator();
```

Comment:
This line causes the build to fail due to a compiler warning -- you need to provide a type parameter for the raw type `IntervalOverlappingIterator`